### PR TITLE
Use `umzug` to autorun migrations

### LINKED ISSUE #5280: Explore `umzug` to autorun migrations
In favor of using `umzug` to [autorun migrations](https://github.com/outline/outline/issues/4680). Much simpler interface compared to what we currently have - `execSync` running commands and then processing output. We can avoid all that and eliminate a bunch of code :relieved:

### PR DESCRIPTION
**Pull request by apoorv-mishra**
Closes #5280

### CODE REVIEW

File: server/utils/startup.ts [hunk 1]
```diff
@@ -1,58 +1,22 @@
...
-}
-
 export async function checkPendingMigrations() {
   try {
-    const pending = getPendingMigrations();
+    const pending = await umzug.pending();
     if (!isEmpty(pending)) {
       if (env.isCloudHosted()) {
-        logMigrations(pending);
+        Logger.warn(chalk.red("Run pending migrations!"));
```
Comment (apoorv-mishra) [inline server/utils/startup.ts:14]: Got rid of `logMigrations`. Wasn't really necessary for the cloud-hosted case I guess.

File: server/database/sequelize.ts [hunk 2]
```diff
@@ -32,3 +34,27 @@ export const sequelize = new Sequelize(url, {
     idle: 10000,
   },
 });
+
+export const umzug = new Umzug({
```
Comment (tommoor) [inline server/database/sequelize.ts:38]:
```suggestion
export const migrations = new Umzug({
```

Lets export this as "migrations"

File: server/database/sequelize.ts [hunk 3]
```diff
@@ -32,3 +34,27 @@ export const sequelize = new Sequelize(url, {
...
+        down: async () => migration.down(context, Sequelize),
+      };
+    },
+  },
+  context: sequelize.getQueryInterface(),
+  storage: new SequelizeStorage({ sequelize }),
+  logger: {
+    warn: (msg) => Logger.warn("database", msg),
+    error: (msg) =>
+      Logger.error("database", new Error("Migration failed!"), msg),
```
Comment (tommoor) [inline server/database/sequelize.ts:56]:
```suggestion
      Logger.error(msg, new Error("Database migration failed")),
```

File: server/utils/startup.ts (hunk 1 above)
Comment (tommoor) [inline server/utils/startup.ts:14]:
```suggestion
        Logger.warn(chalk.red("Migrations are pending"));
```

File: server/utils/startup.ts [hunk 4]
```diff
@@ -1,58 +1,22 @@
...
         process.exit(1);
       } else {
-        runMigrations();
+        Logger.info("database", "Running migrations...");
```
Comment (tommoor) [inline server/utils/startup.ts:17]:
```suggestion
        Logger.info("database", "Running migrations…");
```

File: server/utils/startup.ts [hunk 5]
```diff
@@ -1,58 +1,22 @@
...
+        await umzug.up();
+        Logger.info("database", "Done!");
```
Comment (tommoor) [inline server/utils/startup.ts:19]:
```suggestion
```

**Review by tommoor — COMMENTED** [review]
cool cool

File: server/database/sequelize.ts (hunk 3 above)
Comment (apoorv-mishra) [reply server/database/sequelize.ts:56]: Needs to be stringified...
```suggestion
      Logger.error(JSON.stringify(msg), new Error("Database migration failed")),
```

Comment (tommoor) [reply server/database/sequelize.ts]: What kind of object is `msg` then? I assumed it was a string…

Comment (tommoor) [reply server/database/sequelize.ts]: In that case best passed as the third `extra` argument, first just has to be a repeat of "Database migration failed"

File: server/database/sequelize.ts [hunk 6]
```diff
@@ -32,3 +34,42 @@ export const sequelize = new Sequelize(url, {
...
+    },
+  },
+  context: sequelize.getQueryInterface(),
+  storage: new SequelizeStorage({ sequelize }),
+  logger: {
+    warn: (params) => Logger.warn("database", params),
+    error: (params) =>
+      Logger.error(
+        params.message as string,
+        (params as unknown) as MigrationError
```
Comment (apoorv-mishra) [inline server/database/sequelize.ts:58]: Safe to assume it as [`MigrationError`](https://github.com/sequelize/umzug#errors) :+1:

Comment (tommoor) [reply server/database/sequelize.ts:58]: Probably better off casting in the method params then eg `error: (params: MigrationError) =>` or similar?

Comment (apoorv-mishra) [reply server/database/sequelize.ts]: Better :+1:

**Review by tommoor — APPROVED** [review]

**Pull request merged**